### ISSUE TITLE
Limit django versions in setup.py

### ISSUE THREAD

**Issue by simonsmiley** [django-haystack/django-haystack#1343]
Since the currently released version of haystack is incompatible with django 1.9 people have to manually install django 1.8 it would be nice if pip would handle this automatically.

The same problem will probably also arise when django 1.10 is released.
For the next release please limit django versions to >= 1.8 and < 1.10

I'm not sure when you're planning a new release that supports 1.9. But if it's still a while, can you please release a new version where you just limit the django version?

**Comment by rhblind** [comment]
Any updates on this?
Really want to use Django 1.9 =)

**Comment by acdha** [comment]
Please test the master branch - the release has been waiting on people with time to test & review it.



On Sat, May 7, 2016 at 11:54 PM -0700, "Rolf Håvard Blindheim" <notifications@github.com> wrote:












Any updates on this?

Really want to use Django 1.9 =)

**Comment by claudep** [comment]
Hey Chris, sorry, but I think we are going nowhere asking people to test master. Would it be possible for you to release at least a rc version?

**Comment by rhblind** [comment]
For what it's worth, I've tested the master branch with my [drf-haystack](https://github.com/inonit/drf-haystack) library and it seems to be working good (at least it's passing my test suite). I'm waiting to release a new version until there's a new django-haystack version, so at least an rc version would be fantastic!

**Comment by acdha** [comment]
Since Django 1.10 is nigh, I've added a Travis target which tests the alpha builds but marked it as allow failure. https://travis-ci.org/acdha/django-haystack/jobs/133485740#L1676-L1682 suggests we'll need to deal with the deferred logic changing.

**Comment by simonsmiley** [comment]
@acdha You still didn't fix the original issue. Whenever you install django-haystack it automatically uses the latest version of django. Since it's incompatible with 1.10 all setups from that point in time will fail. You need to limit the django version in setup.py.

A lower and upper limit should be added, to ensure people actually have compatible django versions

**Comment by simonsmiley** [comment]
I created a merge request with the changes I think are required. You'll just need to change the versions whenever a django version is officially supported or dropped

**Comment by acdha** [comment]
@simonsmiley That's what I was using before but it prevented the use of a test target for 1.10

**Comment by simonsmiley** [comment]
@acdha where is this failure?
For testing you should set the django version in tox.ini
I would recommend using the latest archive from django, so you don't have to change your settings for every django (alpha,beta) release

Here is an example of how this should be done (and it works):
[tox.ini](https://gitlab.com/mailman/postorius/blob/master/tox.ini)
[setup.py](https://gitlab.com/mailman/postorius/blob/master/setup.py)

**Comment by simonsmiley** [comment]
I just took a look at tox.ini.
You can improve it:
* squash all the django version definitions
* get rid of all the base python occurrences
* use `{}` in envlist

the tox.ini file I linked above contains all these "improvements"
